src/basic/virt.c

```diff
@@ -484,6 +482,10 @@ int detect_vm(void) {
         else if (r != VIRTUALIZATION_NONE)
                 goto finish;
 
+        /* If we are in Dom0 and have not yet finished, finish with the result of detect_vm_cpuid */
```

Comment:
As detect_vm_cpuid can return VIRTUALIZATION_VM_OTHER and proceed, my previous comment stating that we return VIRTUALIZATION_NONE here was possibly incorrect, so fixed.